python/test_dcr_template.py

```diff
@@ -412,173 +406,142 @@ def setUp(self):
-        data_file = "test_data/build_psf_kernel_full_vals.npy"
-        kernel = self.dcrCorr.build_psf_kernel(use_full=True)
-        kernel_ref = np.load(data_file)
-        self.assertFloatsAlmostEqual(kernel, kernel_ref)
-
-    def test_build_psf_kernel(self):
-        """Compare the result of _build_dcr_kernel to previously computed values."""
-        data_file = "test_data/build_psf_kernel_vals.npy"
-        kernel = self.dcrCorr.build_psf_kernel(use_full=False)
-        kernel_ref = np.load(data_file)
-        self.assertFloatsAlmostEqual(kernel, kernel_ref)
-
     def test_build_dcr_kernel(self):
         """Compare the result of _build_dcr_kernel to previously computed values."""
         data_file = "test_data/build_dcr_kernel_vals.npy"
-        kernel = self.dcrCorr.build_dcr_kernel()
+        kernel = self.dcrCorr.build_dcr_kernel(self.kernel_size)
+        # np.save(data_file, kernel)
         kernel_ref = np.load(data_file)
         self.assertFloatsAlmostEqual(kernel, kernel_ref)
 
     def test_build_model(self):
         """Call build_model with as many options as possible turned off."""
         """Compare the result of build_model to previously computed values."""
         data_file = "test_data/build_model_vals.npy"
-        self.dcrCorr.calc_psf_model()
-        self.dcrCorr.build_model(verbose=False, use_nonnegative=False)
+        self.dcrCorr.build_model(verbose=False)
         model_vals = self.dcrCorr.model
+        # np.save(data_file, model_vals)
         model_ref = np.load(data_file)
-        self.assertFloatsAlmostEqual(model_vals, model_ref)
-
-    def test_solve_model_no_regularization(self):
+        for f, model in enumerate(model_vals):
+            self.assertFloatsAlmostEqual(model, model_ref[f])
+
+    def test_build_matched_template(self):
+        data_file = "test_data/build_matched_template_vals.npy"
+        exposure = self.dcrCorr.exposures[0]
+        self.dcrCorr.build_model(verbose=False)
+        template, variance = self.dcrCorr.build_matched_template(exposure)
+        # np.save(data_file, (template, variance))
+        template_ref, variance_ref = np.load(data_file)
+        self.assertFloatsAlmostEqual(template, template_ref)
+        self.assertFloatsAlmostEqual(variance, variance_ref)
+
+    def test_calculate_new_model(self):
+        data_file = "test_data/calculate_new_model_vals.npy"
+        rand_gen = np.random
+        rand_gen.seed(5)
+        n_step = self.dcrCorr.n_step
+        x_size = self.dcrCorr.x_size
+        y_size = self.dcrCorr.y_size
+        last_solution = [rand_gen.random((y_size, x_size)) for f in range(n_step)]
+        exp_cut = [False for exp_i in range(self.dcrCorr.n_images)]
+        new_solution, inverse_var_arr = self.dcrCorr._calculate_new_model(last_solution, exp_cut)
+        # np.save(data_file, (new_solution, inverse_var_arr))
+        new_solution_ref, inverse_var_arr_ref = np.load(data_file)
+        for f, soln in enumerate(new_solution):
```

Comment:
I suggest you assert that the lists are the same length before comparing them.
Also consider using `itertools.izip` instead of enumerate to avoid the need for an index